api/tests/phones_views_tests.py

```diff
@@ -1156,3 +1173,82 @@ def test_voice_status_completed_reduces_remaining_seconds_to_negative_value(
             "remaining_minutes": 0,
         },
     )
+
+
+def test_voice_status_completed_deletes_call_from_twilio(
+    phone_user, mocked_twilio_client
+):
+    _make_real_phone(phone_user, verified=True)
+    relay_number = _make_relay_number(phone_user, enabled=False)
+    call_sid = "CA1234567890abcdef1234567890abcdef"
+
+    client = APIClient()
+    path = "/api/v1/voice_status"
+    data = {
+        "CallSid": call_sid,
+        "Called": relay_number.number,
+        "CallStatus": "completed",
+        "CallDuration": "27",
+    }
+    mock_call = Mock(spec=["delete"])
+    mocked_twilio_client.calls = Mock(return_value=mock_call)
+
+    response = client.post(path, data, HTTP_X_TWILIO_SIGNATURE="valid")
+
+    assert response.status_code == 200
+    mocked_twilio_client.calls.assert_called_once_with(call_sid)
+    mock_call.delete.assert_called_once()
+
+
+@pytest.mark.django_db
+def test_sms_status_invalid_twilio_signature(mocked_twilio_validator):
+    mocked_twilio_validator.validate = Mock(return_value=False)
+
+    client = APIClient()
+    path = "/api/v1/sms_status"
+    response = client.post(path, {}, HTTP_X_TWILIO_SIGNATURE="invalid")
+
+    assert response.status_code == 400
+    assert "Invalid Signature" in response.data[0].title()
+
+
+def test_sms_status_missing_required_params_error():
+    client = APIClient()
+    path = "/api/v1/sms_status"
+    response = client.post(path, {}, HTTP_X_TWILIO_SIGNATURE="valid")
+    assert response.status_code == 400
+    assert "Missing Smsstatus Or Messagesid" in response.data[0].title()
+
+
+def test_sms_status_before_delivered_does_nothing(mocked_twilio_client):
+    client = APIClient()
+    path = "/api/v1/sms_status"
```

Comment:
Our `end-to-end-local-phone-dev.md` docs are missing configuration for the `voice_status` and `sms_status` endpoints. I probably need additional configuration to get the test working with my own Twilio account, since the status endpoint were not called.

Reply:
I added these to our docs at least.